src/lib/expression/evaluation/expression_functors.hpp

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include "expression_result.hpp"
+
+namespace opossum {
+
+template <typename T>
+constexpr bool is_logical_operand = std::is_same_v<int32_t, T> || std::is_same_v<NullValue, T>;
+
+bool to_bool(const bool value) { return value; }
+bool to_bool(const NullValue& value) { return false; }
+
+template <typename T, typename V>
+T to_value(const V& v) {
+  if constexpr (std::is_same_v<NullValue, V>) {
+    return T{};
+  } else {
+    return v;
+  }
+}
+
+struct TernaryOr {
+  template <typename R, typename A, typename B>
+  struct supports {
+    static constexpr bool value = is_logical_operand<R> && is_logical_operand<A> && is_logical_operand<B>;
+  };
+
+  template <typename R, typename A, typename B>
+  void operator()(R& result_value, bool& result_null, const A& a_value, const bool a_null, const B& b_value,
```

Comment:
For what reason do we have to work with out parameters here?

Reply:
The alternative (struct containing value and null) would require many carefully placed `std::move` to avoid string copying. I found this simpler.